### PR TITLE
Fix issues in naming.md

### LINKED ISSUE #6819: Extraneous content/typos in Polaris naming documentation
## Issue summary

I was reading through https://polaris.shopify.com/foundations/content/naming as part of my onboarding to content design and notice a lot of typos or content that seems to have been strangely concatenated somehow. Screenshots are attached - I would have fixed this myself but in a some cases it wasn't clear was a sentence was supposed to say.

## Expected behavior

Content should be error-free

## Actual behavior

Distracting content that breaks up the guidelines and sometimes makes it hard to understand the guidance.

## Steps to reproduce the problem

1. Navigate to https://polaris.shopify.com/foundations/content/naming
2.  Read documentation and find content errors

01-23-36uqe-swrha
01-25-z0bco-snvsw
01-26-lk9o9-icx74
01-26-v74rm-exmir
01-27-fnmnf-5f7yk
01-28-7x2nj-px3ln
01-29-72z1q-f8xlj

01-17-e05xc-guhna
01-18-knv29-7nfvz
01-19-oo8wz-cmzme
01-20-ceba9-llpj6
01-22-if2j8-qpiva

### PR DESCRIPTION
**Pull request by alex-page**
Fixes https://github.com/Shopify/polaris/issues/6819

### CODE REVIEW

**Comment by laurkim** [comment]
There's some syntax errors in the `### Referring to Shopify and areas in the admin` section. I couldn't figure out how to add a suggestion comment since it wasn't included in the diff 😓 

`line 286`:
> Descriptive feature names aren’t [capitalized](/content/naming#section-general-guidel featuren providing steps in a workflow, it’s okay to capitalize the page name, for example, “Go to the Products page”. Note that the page name is capitalized, but “page” isn’t. forn referring to Shopify’s main product offering, use “Shopify”. The only exception is when you need to differentiate it from another product, like the butile app, or expl fornpify admin”.

suggestion:
```
Descriptive feature names aren’t [capitalized](/content/naming#section-general-guidelines), but when providing steps in a workflow, it’s okay to capitalize the page name, for example, “Go to the Products page”. Note that the page name is capitalized, but “page” isn’t. When referring to Shopify’s main product offering, use “Shopify”. The only exception is when you need to differentiate it from another product, like the mobile app, or explain a task specific for admin.
```

`line 288-290`:
``` 
A good description can: exception mobileease adoption of the product area

- Help establish a mental model for merchants
```

suggestion:
```
A good description can: 

- Increase adoption of the product area
- Help establish a mental model for merchants
```

File: polaris.shopify.com/content/foundations/content/naming.md [hunk 1]
```diff
@@ -81,16 +81,15 @@ Only Shopify can use the word “Shopify” in a name. The word “Shopify” ca
...
+Adding “Shopify” doesn’t add clarity in the context of other Shopify products and features. For example, merchants often confuse Shopify Shipping with the other shipping features, like shipping settings, carrier calculated shipping rates, shipping labels, and shipping zones. Our support staff have to refer to the Shopify Plan by its cost because the general name doesn’t distinguish it from the other plans.
 
 Use “Shopify” in front of a name when a product:
 
 - Is or will become a separate product or platform and we need to associate it with Shopify
 - Should be differentiated from other, similar products in the industry
-- Doesn’t justify an
-  [evocative name](/content/naming#section-descriptive-vs-evocative-names), but still needs to associated with Shopify
+- Doesn’t justify an [evocative name](/content/naming#section-descriptive-vs-evocative-names), but still needs to associated with Shopify
```
Comment (laurkim) [inline polaris.shopify.com/content/foundations/content/naming.md]:
```suggestion
- Doesn’t justify an [evocative name](/content/naming#descriptive-vs-evocative-names), but still needs to associated with Shopify
```

Ran it locally and looks like links to other markdown sections don't work if you include `section-`

**Comment by laurkim** [comment]
`line 166`:
> If it’s a default feature (merchants don’t have to sign up or opt in to use it), don’t [capitalize](/content/naming#section-general-guidelines) it. don- dodont -->

suggestion:
```
If it’s a default feature (merchants don’t have to sign up or opt in to use it), don’t [capitalize](/content/naming#general-guidelines) it. 

<!-- dodont -->
```


`line 184`:
```
<!-- dodont --> when
```

suggestion:
```
<!-- dodont -->
```

`line 209`:
```
[guidelines for using “Shopify” in a name](/content/naming#section-does-it-need-a-branded-name-).
```

suggestion:
```
[guidelines for using “Shopify” in a name](/content/naming#does-it-need-a-branded-name-).
```


`line 236`:
> Some evocative names can be more descriptive, although they’re harder to trademark if they use common terminology. Not all evocative names need t trademarked.

suggestion:
```
Some evocative names can be more descriptive, although they’re harder to trademark if they use common terminology. Not all evocative names need to be trademarked.
```

File: polaris.shopify.com/content/foundations/content/naming.md [hunk 2]
```diff
@@ -279,10 +282,11 @@ There are two types of approaches to naming, the
 
 ### Referring to Shopify and areas in the admin
 
-Use consistent and easy to understand descriptions when referring to locations in product, especially in
-[help documentation](/content/help-documentation#navigation). Descriptive feature names aren’t [capitalized](/content/naming#section-general-guidel featuren providing steps in a workflow, it’s okay to capitalize the page name, for example, “Go to the Products page”. Note that the page name is capitalized, but “page” isn’t. forn referring to Shopify’s main product offering, use “Shopify”. The only exception is when you need to differentiate it from another product, like the butile app, or expl fornpify admin”.
+Use consistent and easy to understand descriptions when referring to locations in product, especially in [help documentation](/content/help-documentation#navigation). Descriptive feature names aren’t [capitalized](https://polaris.shopify.com/foundations/content/naming), but when providing steps in a workflow, it’s okay to capitalize the page name, for example, “Go to the Products page”. Note that the page name is capitalized, but “page” isn’t.
```
Comment (laurkim) [inline polaris.shopify.com/content/foundations/content/naming.md]:
```suggestion
Use consistent and easy to understand descriptions when referring to locations in product, especially in [help documentation](/content/help-documentation#navigation). Descriptive feature names aren’t [capitalized](/content/naming#general-guidelines), but when providing steps in a workflow, it’s okay to capitalize the page name, for example, “Go to the Products page”. Note that the page name is capitalized, but “page” isn’t.
```

changed ref for capitalized link

**Review by laurkim — APPROVED** [review]
Added some small suggestions in the comment above. 

Don't think all of them showed up in the diff, but the links for `(/content/naming#section-XX)` just need to be updated to remove `section-` to be able to link to the correct sections.

**Comment by alex-page** [comment]
@laurkim https://github.com/Shopify/polaris/pull/6826 addresses the anchor links.

**Comment by jpat90** [comment]
Thanks for jumping on this, @alex-page!

**Pull request merged**